vm/interpreter/src/vm.rs

```diff
@@ -354,3 +357,56 @@ fn check_message(msg: &Message) -> Result<(), anyhow::Error> {
 
     Ok(())
 }
+
+/// Default reward working with the Filecoin Reward Actor.
+pub struct RewardActorMessageCalc;
+
+impl RewardCalc for RewardActorMessageCalc {
+    fn reward_message(
+        &self,
+        epoch: ChainEpoch,
+        miner: Address,
+        win_count: i64,
+        penalty: BigInt,
+        gas_reward: BigInt,
+    ) -> Result<Option<Message>, fvm_ipld_encoding::Error> {
+        let params = Serialized::serialize(AwardBlockRewardParams {
+            miner,
+            penalty,
+            gas_reward,
+            win_count,
+        })?;
+
+        let rew_msg = Message {
+            from: system::ADDRESS,
+            to: reward::ADDRESS,
+            method_num: reward::Method::AwardBlockReward as u64,
+            params,
+            // Epoch as sequence is intentional
+            sequence: epoch as u64,
+            gas_limit: 1 << 30,
+            value: Default::default(),
+            version: Default::default(),
+            gas_fee_cap: Default::default(),
+            gas_premium: Default::default(),
+        };
+
+        Ok(Some(rew_msg))
+    }
+}
+
+/// Not giving any reward for block creation.
+pub struct NoRewardCalc;
+
+impl RewardCalc for NoRewardCalc {
```

Comment:
So in your example instead of sending the gas/penalty from the system actor to the reward actor, you send some reward value from the reward actor to the beneficiary directly.

I didn't know this was the way to do it, thank you for pointing it out; I will add this example as well. 

With regards to whether 1 FIL or no-reward should be the default in Delegated Consensus, I thought no reward is not a bad option either because if there's only 1 delegated miner, and nobody has a chance to take over, then why should they be rewarded, or rather, why should everybody else _not_ be rewarded? They only service this one is doing is the selection of messages from the mempool. And why 1 FIL, why not a configurable value or one driven by governance? I thought no reward is a default that invites less questions. But if you just want to demonstrate technically that rewards are given, 1 FIL is as good as any other value for sure.

Reply:
I see that actually you [add 1 FIL to the gasReward](https://github.com/filecoin-project/eudico/blob/eudico/chain/consensus/common/executor.go#L158) and ignore penalties. This is more non-standard than a fixed reward: I understood that you get 1 FIL and no reward for gas, so fees are effectively burned.

Okay then I will implement this under delegcns, rather than add a fixed example, which I'm not sure is useful on its own.

Reply:
Done in https://github.com/ChainSafe/forest/pull/1766/commits/2bed7a5b876d060d161fe64e49b8ffad3be5c6a9

Put in the VM after all.